app/controllers/suggestions.rb

```diff
@@ -0,0 +1,75 @@
+get '/suggestions' do
+  @suggestions = Suggestion.all
+  @suggestions_by_votecount = @suggestions.sort_by{|suggestion| suggestion.votes.count}.reverse
+  erb :'suggestions/index'
+end
+
+get '/suggestions/new' do
+  @suggestion = Suggestion.new
+  erb :'suggestions/new'
+end
+
+post '/suggestions' do
+  @suggestion = current_user.suggestions.create(params[:suggestion])
+  redirect '/suggestions'
+end
+
+get '/suggestions/:id' do
+  @suggestion = Suggestion.find_by(id: params[:id])
+  erb :'suggestions/show'
+end
+
+get '/suggestions/:id/edit' do
+  @suggestion = Suggestion.find_by(id: params[:id])
+  if current_user == @suggestion.user
+    erb :'suggestions/edit'
+  else
+    status 401
+    redirect "/suggestions/#{@suggestion.id}"
+  end
+end
+
+patch '/suggestions/:id' do
+  puts params
+  @suggestion = Suggestion.find_by(id: params[:id])
+  @suggestion.update(params[:suggestion])
+  redirect "/suggestions"
+end
+
+delete '/suggestions/:id' do
+  @suggestion = Suggestion.find_by(id: params[:id])
+  if current_user == @suggestion.user
+    Suggestion.find(params[:id]).destroy!
+    redirect '/suggestions/:id'
+  else
+    status 401
+    redirect '/suggestions/:id'
```

Comment:
`redirect` needs the actual path passed to it. Thus `:id` is not going to work in these paths. Since this is a redirect after a delete, probably best to redirect to either suggestions index or user profile